src/utils/entitySpec.js

```diff
@@ -15,25 +15,47 @@ const toLowerCaseMap = {
   mavencentralsource: NONE
 }
 
-const NPM_WEBSITE = 'npmjs.com'
-const GITHUB_WEBSITE = 'github.com'
-const MAVEN_WEBSITE = 'mvnrepository.com'
-const NUGET_WEBSITE = 'nuget.org'
-const PYPI_WEBSITE = 'pypi.org'
-const RUBYGEM_WEBSITE = 'rubygems.org'
-
-const providerPath = {
-  [NPM_WEBSITE]: 'npm/npmjs',
-  [GITHUB_WEBSITE]: 'git/github',
-  [PYPI_WEBSITE]: 'pypi/pypi/-',
-  [MAVEN_WEBSITE]: 'maven/mavencentral',
-  [NUGET_WEBSITE]: 'nuget/nuget/-',
-  [RUBYGEM_WEBSITE]: 'gem/rubygems/-'
+const entityMapping = [
+  { hostnames: ['npmjs.com', 'npmjs.org'], parser: npmParser },
+  { hostnames: ['github.com'], parser: githubParser },
+  { hostnames: ['mvnrepository.com'], parser: mavenParser },
+  { hostnames: ['nuget.org'], parser: nugetParser },
+  { hostnames: ['pypi.org'], parser: pypiParser },
+  { hostnames: ['rubygems.org'], parser: rubygemsParser }
+]
+
+function npmParser(host, path) {
+  let namespace, name, version
+  const pathSegments = path.split('/')
+  if (pathSegments.length === 5) [, namespace, name, , version] = pathSegments
+  else [, name, , version] = pathSegments
+  return new EntitySpec('npm', 'npmjs', namespace, name, version)
 }
 
-const providerWebsiteValues = [NPM_WEBSITE, GITHUB_WEBSITE, MAVEN_WEBSITE, NUGET_WEBSITE, PYPI_WEBSITE, RUBYGEM_WEBSITE]
+function githubParser(host, path, error) {
```

Comment:
How `error` is used here?
I saw this in other methods, but it seems to be never used

Reply:
that was stale and is now removed